magda-io/magda
# Added auth to links from the Registry /records (plural) endpoint

### What this PR does

Fixes #2371 

This adds auth for links to the `/records` endpoint (previously we implemented only `records/<recordid>`. So essentially, when you call `/records` on the registry, inner records should appear or not appear based on the policy set on those records.

To test:
Go to https://issues-2371-links-in-records-plural.dev.magda.io/api/v0/registry/records?aspect=aspect-with-link-array while logged out, you should see that `record-outer-multiple-public` has two inner records, `record-outer-multiple-mixed` has one inner record, and `record-outer-multiple-none-allowed` has no inner records. If you log in and call it again, all should have two inner records.

Similarly, if you go to https://issues-2371-links-in-records-plural.dev.magda.io/api/v0/registry/records?aspect=aspect-with-link-single while logged out you should see that `record-outer-single-public` has an inner record and `record-outer-single-private` has no inner record. Sign in and they should both have an inner record.

### Checklist
